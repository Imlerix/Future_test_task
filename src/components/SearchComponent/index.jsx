import React from 'react';
import style from './style.module.css';
import PropTypes from 'prop-types';

import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Toolbar from "@material-ui/core/Toolbar";

class SearchComponent extends React.Component{
    constructor(props){
        super(props)
        this.state = {}
    }

    onChangeInput = (e) => {
        const value = e.target.value;
        this.props.listener(value);
    }

    render() {
        return(
            <div>
                <TextField onChange={this.onChangeInput}
                           disabled={this.props.disabled}
                           value={this.props.value}
                           label="Поиск"/>
            </div>
        )
    }
}

SearchComponent.propTypes = {
    value: PropTypes.string.isRequired,
    listener: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
};

export default SearchComponent;
