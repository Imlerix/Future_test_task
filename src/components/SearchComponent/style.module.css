.SearchComponentContainer{
    /*display: flex;*/
    /*align-items: flex-end;*/
    /*width: 100%;*/
    /*padding: 0 20px 10px 18px;*/
    flex: 1 1 100%;
}
.SearchComponent_functionalHead_searchBtn{
    margin: 0 0 0 10px;
}
