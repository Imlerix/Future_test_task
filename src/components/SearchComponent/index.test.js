import renderer from 'react-test-renderer'
import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import SearchComponent from './index';

describe('>>>SearchComponent --- Snapshot',()=>{
    it('+++capturing Snapshot of SearchComponent', () => {
        const renderedValue =  renderer.create(
            <SearchComponent disabled={false}
                             value={''}
                             listener={() => {}}/>
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

////
let container = null;
beforeEach(() => {
    // подготавливаем DOM-элемент, куда будем рендерить
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // подчищаем после завершения
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("renders correct text in input", () => {
    act(() => {
        render(
            <SearchComponent disabled={true}
                             value="Ввод в инпут"
                             listener={() => {}}/>
            ,container);
    });
    expect(container.querySelector('input').value).toBe("Ввод в инпут");
});
