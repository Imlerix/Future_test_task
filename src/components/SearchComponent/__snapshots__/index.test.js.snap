// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SearchComponent --- Snapshot +++capturing Snapshot of SearchComponent 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Поиск
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
